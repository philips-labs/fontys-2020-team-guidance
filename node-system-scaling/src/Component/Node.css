.nodeImg {
    -webkit-user-drag: none;
    -webkit-user-select: none;
    width: 2.5vw;
    height: auto;
}



.nodeSmallCircle {
    display: block;
    --main-height: 75px;
    width: 75px;
    line-height: var(--main-height);

    -moz-border-radius: 50%;
    border-radius: 100vw;
    border: 2px solid black;

    background-color: lightgreen;
    opacity: 50%;
    color: white;
    text-align: center;
    font-size: 2em;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}

.nodeMedCircle {
    display: block;
    --main-height: 100px;
    width: 100px;
    line-height: var(--main-height);

    -moz-border-radius: 50%;
    border-radius: 100vw;
    border: 2px solid black;

    background-color: orange;
    opacity: 50%;
    color: white;
    text-align: center;
    font-size: 2em;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}

.nodeBigCircle {
    display: block;
    --main-height: 150px;
    width: 150px;
    line-height: var(--main-height);

    -moz-border-radius: 50%;
    border-radius: 100vw;
    border: 2px solid black;


    background-color: red;
    opacity: 50%;
    color: white;
    text-align: center;
    font-size: 2em;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}

.draggingBounds {
    height: 600px;
    width: auto;
    margin-top: 99px;
    margin-left: 421px;
    margin-right: 421px;
}

.lockNodes {
    pointer-events: none
}

.node {
    height: 46.42px;
    width: 46.42px !important;
    float: left;
}

.linkMenu {
    float: left;
    width: 400px;
}