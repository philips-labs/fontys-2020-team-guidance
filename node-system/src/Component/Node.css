.nodeImg {
    -webkit-user-drag: none;
    -webkit-user-select: none;
    width: 4vw;
    height: auto;
}