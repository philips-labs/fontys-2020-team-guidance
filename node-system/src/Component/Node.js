import React, {Component} from 'react';
import Waypoint from './Images/waypoint.png';
import './Node.css';

export default class Node extends Component {
    constructor(prop) {
        super(prop);
        this.state = {
            connectedNodes: [],
            nodeData: this.props.data
        };
    }

    getStyle() {
        let connections;
        let data = this.state.nodeData;

        let nodeConnections = data.map((conn) => {
            if(conn.id === this.props.nodeId) {
                connections = conn.connections
            }
            return connections
        })

        if(connections < 5) {
            return 'nodeSmallCircle'
        }
        else if(connections >= 5 && connections < 10){
            return 'nodeMedCircle'
        }
        else if(connections >= 10) {
            return 'nodeBigCircle'
        }
        else return ''
    }

    render() {
        return (
            <div className={this.getStyle()}>
                <img className='nodeImg' src= {Waypoint} alt=''/>
            </div>

        )
    }
}