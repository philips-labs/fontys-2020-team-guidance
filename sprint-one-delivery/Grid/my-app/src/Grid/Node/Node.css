.node {
    width: 7px;
    height: 7px;
    border: 0.05px solid rgb(124, 155, 255);
    float: left;
}

div.node:hover {
    background-color: lightblue;
}

.nothing {
    background-color: transparent ;
}

.wall {
    background-color: black ;
}

.startingpoint {
    background-color: greenyellow ;
}

.endpoint {
    background-color: red ;
}

.path {
    background-color: lightseagreen ;
}