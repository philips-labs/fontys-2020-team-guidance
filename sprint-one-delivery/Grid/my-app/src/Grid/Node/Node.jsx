import React, {Component} from 'react';
import './Node.css';

class Node extends Component {
    constructor(props) {
        super(props);
        this.state = {obj: "nothing",colorId: "0"};
    
        // This binding is necessary to make `this` work in the callback
        this.setColor = this.setColor.bind(this);
      }
    
      setColor(e) {
          if (this.state.colorId === "0")
          {
            this.setState({obj: "wall",colorId: "1"});
          }
          else if (this.state.colorId === "1")
          {
            this.setState({obj: "startingpoint",colorId: "2"});
          }
          else if (this.state.colorId === "2")
          {
            this.setState({obj: "endpoint",colorId: "3"});
          }
          else{
            this.setState({obj: "nothing",colorId: "0"});
          }
        }

    render() {
        let id = this.state.colorId;
        let obj = this.state.obj;
        console.log(id);
        console.log(obj);
        return (
        <div className={obj + " node"} id={id} onClick={this.setColor.bind(this)}></div>
        );
    }
}
export default Node;
