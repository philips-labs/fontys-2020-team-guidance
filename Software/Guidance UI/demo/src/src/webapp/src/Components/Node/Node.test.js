import React from 'react';
import { render, cleanup, fireEvent} from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Node from "./Node";

afterEach(cleanup);

it('renders correctly react-test-renderer', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Node />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});