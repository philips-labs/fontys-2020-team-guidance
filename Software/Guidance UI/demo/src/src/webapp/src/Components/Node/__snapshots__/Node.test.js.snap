// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `
<div>
  <img
    alt=""
    className="nodeImg"
    src="waypoint.png"
  />
  <p />
</div>
`;
